crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -101,18 +101,21 @@ impl std::fmt::Display for ModuleMember {
 /// Returns the PEP 585 standard library generic variant for a `typing` module reference, if such
 /// a variant exists.
 pub fn to_pep585_generic(expr: &Expr, semantic: &SemanticModel) -> Option<ModuleMember> {
-    semantic.resolve_call_path(expr).and_then(|call_path| {
-        let [module, member] = call_path.as_slice() else {
-            return None;
-        };
-        as_pep_585_generic(module, member).map(|(module, member)| {
-            if module.is_empty() {
-                ModuleMember::BuiltIn(member)
-            } else {
-                ModuleMember::Member(module, member)
-            }
+    Some(expr)
+        .filter(|_| semantic.seen(Modules::TYPING | Modules::TYPING_EXTENSIONS))
+        .and_then(|expr| semantic.resolve_call_path(expr))
```

Comment:
This reads a bit inside-out (`Some(expr)`)

You can use `bool.then` or `bool.then_some`

```suggestion
semantic.seen(Modules::TYPING | Modules::TYPING_EXTENSIONS)
	.then(||semantic.resolve_call_path(expr))
```